gatsby-node.js

```diff
@@ -289,18 +289,57 @@ async function getIndexToUse({ index, tempIndex, enablePartialUpdates }) {
   return index;
 }
 
-async function getSettingsToApply({ settings, index, tempIndex, indexToUse }) {
-  const requestedSettings = settings ? settings : await index.getSettings();
+
+async function getSettingsToApply({
+  settings: givenSettings,
+  index,
+  tempIndex,
+  indexToUse,
+}) {
+  const { replicaUpdateMode, ...settings } = givenSettings;
```

Comment:
@yogainternational and @Haroenv 

Due to this change, if `replicaUpdateMode` is not specifically set on `options.settings` it triggers an error saying:

`TypeError: Cannot destructure property `replicaUpdateMode` of 'undefined' or '  null'.`

Is this the desired behaviour? 
Requiring `replicaUpdateMode` on settings?

Reply:
ah that's an important thing to note, will fix @crsantos

Reply:
Hmm @crsantos, which settings do you use? can you send the configuration? I think this should already have been guarding 🤔

Reply:
I didn't have any, so that's why I had that error 😄

Reply:
ahh yes of course, thanks! fixed in 0.14.1 @crsantos :)

Reply:
This was my `gatsby-config.js` part of Algolia plugin:

```jsx
plugins: [
    `gatsby-plugin-styled-jsx`, // the plugin's code is inserted directly to gatsby-node.js and gatsby-ssr.js files
    `gatsby-plugin-styled-jsx-postcss`, // as above
    {
      resolve: `gatsby-plugin-layout`,
      options: {
        component: require.resolve(`./src/layouts/`)
      }
    },
    {
      resolve: `gatsby-plugin-algolia`,
      options: {
        appId: process.env.ALGOLIA_APP_ID ? process.env.ALGOLIA_APP_ID : "",
        apiKey: process.env.ALGOLIA_ADMIN_API_KEY ? process.env.ALGOLIA_ADMIN_API_KEY : "",
        indexName: process.env.ALGOLIA_INDEX_NAME ? process.env.ALGOLIA_INDEX_NAME : "",
        queries,
        settings: {
          // I had nothing here, so I added : replicaUpdateMode: "replace"
        },
        chunkSize: 10000 // default: 1000
      }
    }
```